pkg/steps/run.go

```diff
@@ -56,7 +56,9 @@ func Run(ctx context.Context, graph []*api.StepNode) (*junit.TestSuites, []api.C
 			stepDetails = append(stepDetails, out.stepDetails)
 			if out.err != nil {
 				testCase.FailureOutput = &junit.FailureOutput{Output: out.err.Error()}
-				executionErrors = append(executionErrors, results.ForReason("step_failed").WithError(out.err).Errorf("step %s failed: %v", out.node.Step.Name(), out.err))
+				if out.err != context.Canceled {
```

Comment:
Can you add a testcase for this so we don't regress?

Reply:
Well, it seems that we don't even have tests for this. Sounds like a good opportunity to write some.

Reply:
Should this be `&& interrupted ` or am I missing something? I think there are other causes for cancellations, and I thnk we're interested in them.

Reply:
We check this in the above case.

Reply:
How are we checking for `interrupted`?

Reply:
we are passing already a context.WithCancel, so this will be checked in the `context.Done()` case

Reply:
Hmm I guess so the issue here is that we cannot tell *why* we were interrupted - was it a timeout or kubelet killing the pod or whatever. I don't think there's any way around that. SGTM